recordset.go

```diff
@@ -102,6 +103,24 @@ func (rcs *Recordset) IsActive() bool {
 	return rcs.active.Get()
 }
 
+var ErrRecordsetClosed = errors.New("aerospike: Recordset closed")
+
+// Read reads the next record from the Recordset. If the Recordset has been
+// closed, it returns ErrRecordsetClosed.
+func (rcs *Recordset) Read() (record *Record, err error) {
+	var ok bool
+	select {
+	case <-rcs.cancelled:
+	case record, ok = <-rcs.Records:
+	case err, ok = <-rcs.Errors:
```

Comment:
Thank you for your suggestions. You are correct that there is a potential goroutine leak in that method (it can be worse since the resources for the command won't be freed as well).

Interestingly, the problem is not as simple as it seems. For example, here you should not check if the error channel is closed, since if both chans are closed (happens in `Close` method), some records may be left in the `Records` chan. Please take a look at the following example to see how ```close(chan)``` works: https://play.golang.org/p/XRnLcmxeBL
There's also no need to directly read `cancelled` channel since it will automatically close the `Records` and `Errors` chans in time from the worker commands.

More troubling is that there's no way of automatically deallocating the resources in case the user forgets to do so, without consuming all records or manually calling the ```Close``` method. 

There's also the issue of the error type for the end of recordset which is not compatible with the generic ```AerospikeError```.

I see your point though, and will take care of the required changes for the next release, including introduction of the new ```Read()``` method.